src/Elasticsearch.Net/Configuration/IConnectionConfigurationValues.cs

```diff
@@ -213,5 +214,11 @@ public interface IConnectionConfigurationValues : IDisposable
 		bool ThrowExceptions { get; }
 
 		ElasticsearchUrlFormatter UrlFormatter { get; }
+
+		/// <summary>
+		/// The user agent string to send with requests. Useful for debugging purposes to understand client and framework
+		/// versions that initiate requests to Elasticsearch
+		/// </summary>
+		string UserAgent { get; }
```

Comment:
Might be worth including a few examples?

Reply:
What would those examples look like? Does providing examples encourage its usage? If it would, I don't think we should provide any.

There's the [RFC 2616 for Hypertext Transfer Protocol -- HTTP/1.1, that includes User-Agent](https://tools.ietf.org/html/rfc2616#page-145), which Elasticsearch may or may not adhere to (would need to dig into). 

The value I see in providing a way to set your own value is only really for the case where the default value that the client would now send messing with the environment in some way e.g. a proxy in between the client and Elasticsearch that doesn't like the default value.

Reply:
I was just thinking from the perspective of the user who can read the code, see an example of a user agent string and say to themselves "Oh, I get it".

Its a `{ get; }` anyway.